lib/config.js

```diff
@@ -1653,17 +1673,25 @@ util.runStrictnessChecks = function (config) {
     _warnOrThrow("NODE_ENV value of '"+NODE_ENV+"' is ambiguous.");
   }
 
-  function _warnOrThrow (msg) {
-    var beStrict = process.env.NODE_CONFIG_STRICT_MODE;
-    var prefix = beStrict ? 'FATAL: ' : 'WARNING: ';
-    var seeURL = 'See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+  function _warnOrThrow (msg, neverThrow) {
+    var prefix = beStrict && !neverThrow ? 'FATAL: ' : 'WARNING: ';
 
     console.error(prefix+msg);
-    console.error(prefix+seeURL);
 
-    // Accept 1 and true as truthy values. When set via process.env, Node.js casts them to strings.
-    if (["true", "1"].indexOf(beStrict) >= 0) {
-      throw new Error(prefix+msg+' '+seeURL);
+    // Unless this is a warning that's never fatal, let the user know about strict mode.
+    if (!neverThrow) {
+      var modeInfo = beStrict ? 'Strict mode is on, aborting.' : 'Strict mode is off, continuing.';
+      var modeMessage =  modeInfo + ' See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+      console.error(prefix+modeMessage);
+      if (beStrict) {
+        throw new Error(prefix+msg+' '+modeMessage);
+      }
+    }
+
+    // If we're not in strict mode or this is a message that's safe to disable, let the user
+    // know that they can turn off warnings altogether
+    if (!beStrict || neverThrow) {
+      console.error(prefix+'To disable this warning, set SUPPRESS_NO_CONFIG_WARNING in the environment.');
```

Comment:
I don't understand what you mean by "you're outputting advise for only one of the invocations". The "To disable this warning, ..." message is shown for *all* invocations if strict mode is off (since in those cases, it would have an effect), and independent of strict mode for the single case of "no config file", since it can be safely dismissed with strict mode on in that case (unchanged behavior).